spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -952,32 +947,23 @@ object QueryPlanSerde extends Logging with CometExprShim {
           binding,
           (builder, binaryExpr) => builder.setRlike(binaryExpr))
 
-      case StartsWith(left, right) =>
-        createBinaryExpr(
-          expr,
-          left,
-          right,
-          inputs,
-          binding,
-          (builder, binaryExpr) => builder.setStartsWith(binaryExpr))
-
-      case EndsWith(left, right) =>
-        createBinaryExpr(
-          expr,
-          left,
-          right,
-          inputs,
-          binding,
-          (builder, binaryExpr) => builder.setEndsWith(binaryExpr))
-
-      case Contains(left, right) =>
-        createBinaryExpr(
-          expr,
-          left,
-          right,
-          inputs,
-          binding,
-          (builder, binaryExpr) => builder.setContains(binaryExpr))
+      case _: StartsWith =>
+        val startsWithExpr = expr.asInstanceOf[StartsWith]
```

Comment:
The `asInstanceOf` cast can be avoided here:

```suggestion
      case startsWithExpr: StartsWith =>
```